### ISSUE TITLE
Unsupported authentication method: LDAP

### ISSUE THREAD

**Issue by dockerzhang** [yaooqinn/kyuubi#128]
**Describe the bug**
In this https://yaooqinn.github.io/kyuubi/docs/authentication.html documentation,  it says that authentication option include LDAP, but it will get ServiceException when set spark.kyuubi.authentication=LDAP.
yaooqinn.kyuubi.service.ServiceException: Unsupported authentication method: LDAP
        at yaooqinn.kyuubi.auth.AuthType$.toAuthType(AuthType.scala:46)
        at yaooqinn.kyuubi.auth.KyuubiAuthFactory.<init>(KyuubiAuthFactory.scala:46)
        at yaooqinn.kyuubi.server.FrontendService.run(FrontendService.scala:585)
        at java.lang.Thread.run(Thread.java:745)
2018-12-02 16:33:06 [INFO] Shutting down KyuubiServeryaooqinn.kyuubi.Logging$class.info(Logging.scala:33)

**To Reproduce**
--conf spark.kyuubi.authentication=LDAP

**Expected behavior**
support LDAP authentication

**Screenshots**
![image](https://user-images.githubusercontent.com/18047329/49337674-c7d1cf00-f651-11e8-9be7-1d0c651c60e4.png)

**Additional context**

**Comment by yaooqinn** [comment]
nice catch, its my mistake to forget add ldap auth to kyuubi. let me know if you have interest in fixing it, or I will do as soon as possible.

**Comment by dockerzhang** [comment]
@yaooqinn Thank you very much for your PR. I tested KYUUBI-128 in my environment, there are two issues.
1, I got "Unsupported authentication method: LDAP" exception in KyuubiAuthFactory.scala(from 69 line).  After i added "AuthType.LDAP" , it can run successfully.
![image](https://user-images.githubusercontent.com/18047329/49784377-0bb28b80-fd58-11e8-95cb-ad0ff1b24e7a.png)
2, I connected kyuubi server by beeline（with "-n admin -p zhang" options）, I found user and password were transferred to ASCII string.
![image](https://user-images.githubusercontent.com/18047329/49785287-086ccf00-fd5b-11e8-8134-bb864d4c1374.png)
Could we talk the detail on QQ, my QQ ID is 296654125？
